homeassistant/components/climate/hive.py

```diff
@@ -0,0 +1,224 @@
+"""
+Support for the Hive devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/hive/
+"""
+import logging
+from datetime import datetime
+
+from homeassistant.components.climate import (ClimateDevice,
+                                              STATE_AUTO, STATE_HEAT,
+                                              STATE_OFF, STATE_ON)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.components.hive import DATA_HIVE
+
+DEPENDENCIES = ['hive']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, hivedevice, discovery_info=None):
+    """Set up Hive climate devices."""
+    session = hass.data.get(DATA_HIVE)
+
+    add_devices([HiveClimateEntity(hass, session, hivedevice)])
+
+
+class HiveClimateEntity(ClimateDevice):
+    """Hive Climate Device."""
+
+    def __init__(self, hass, Session, HiveDevice):

```

Comment:
Don't pass in `hass`. It will be set on the entity when it has been added to home assistant. Also, don't capitalize `Session`. It's an instance and not a class. The same goes for `HiveDevice`.

Reply:
will rename the capitalised names.
how do I reference hass if I don't pass it?

Reply:
done

Reply:
You can use `self.hass` in all methods after the entity has been added to home assistant.

Reply:
ok. i noticed after checking that needed to access hass there was from some legacy code that has been removed